### PR TITLE
ci(circle): add react node cache key suffix

### PR DESCRIPTION
**Pull request by loksonarius**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by dzhu — APPROVED** [review]
I would say the cache buster should stay in the raw key for visibility and to match the other cache jobs. Also maybe additionally print out `$UID` and `$HOME` to the console for the sake of debugging. Otherwise, looks good, thanks!

File: .circleci/config.yml [hunk 1]
```diff
@@ -276,15 +276,21 @@ commands:
     steps:
       - attach_workspace:
           at: .
+      - run:
+          name: Write cache suffix
+          command: |
+            echo '<<pipeline.parameters.cache-buster>>' > /tmp/react-get-deps.cachekey
+            cat webui/react/package-lock.json >> /tmp/react-get-deps.cachekey
+            echo $UID $HOME >> /tmp/react-get-deps.cachekey
```
Comment (dannysauer) [inline .circleci/config.yml:284]: Are there $UIDs which share a $HOME? Feels like one or the other should be sufficiently unique. :)

Comment (dzhu) [reply .circleci/config.yml:284]: Oh right, this should be the current directory. I started off thinking of using home directory, then switched to current directory, but forgot to update some parts of my brain.

**Comment by EmilyBonar** [comment]
Anything still holding this up?

**Comment by loksonarius** [comment]
Sorry, this fell off my plate while I was doing some clerical work.
Addressed comments in a new commit.
Enabling auto-merge -- this should go in ASAP.

**Pull request merged**